File: tests/test_translate_object_bonly_es6/try_except_simple.js
var value;
function MyError(message) {
    this.name = "MyError";
    this.message = (message || "Custom error MyError");
    if (((typeof Error.captureStackTrace) === "function")) {
        Error.captureStackTrace(this, this.constructor);
    } else {
        this.stack = new Error(message).stack;
    }
}
MyError.prototype = Object.create(Error.prototype);
MyError.prototype.constructor = MyError;
class MySecondError extends MyError {
    /* A stupid error */
}
try {
    value = 0;
    throw new MySecondError("This is an error");
} catch(e) {
    if ((e instanceof MySecondError)) {
        value = 1;
    } else {
        throw e;
    }
}

File: tests/test_various/test_translate_object/try_except_simple.py
## body_only: True
## enable_es6: True

def func():

    class MyError(Exception):
        pass

    class MySecondError(MyError):
        """A stupid error"""

    try:
        value = 0
        raise MySecondError('This is an error')
    except MySecondError:
        value = 1
